packages/zoe/test/unitTests/contracts/escrowToVote.js

```diff
@@ -0,0 +1,121 @@
+/* global harden */
+// @ts-check
+
+import { assert, details, q } from '@agoric/assert';
+import makeStore from '@agoric/store';
+// Eventually will be importable from '@agoric/zoe-contract-support'
+import { makeZoeHelpers } from '../../../src/contractSupport';
+
+/**
+ * This contract implements coin voting. There are two roles: the
+ * Secretary, who can determine the question (a string), make voting
+ * invites, and close the election; and the Voters, who can vote YES or
+ * NO on the question. The voters can only get the capability to vote
+ * by making an offer using a voter invite and escrowing assets. The
+ * brand of assets is determined on contract instantiation through an
+ * issuerKeywordRecord. The instantiator gets the only Secretary
+ * invite.
+ *
+ * @typedef {import('../../../src/zoe').ContractFacet} ContractFacet
+ * @typedef {import('@agoric/ERTP').Amount} Amount
+ * @param {ContractFacet} zcf
+ */
+const makeContract = zcf => {
+  const { assertKeywords, checkHook } = makeZoeHelpers(zcf);
+  assertKeywords(harden(['Assets']));
+  const {
+    terms: { question },
+    brandKeywordRecord: { Assets: assetsBrand },
+  } = zcf.getInstanceRecord();
+  assert.typeof(question, 'string');
+  const amountMath = zcf.getAmountMath(assetsBrand);
+
+  const offerHandleToResponse = makeStore('offerHandle');
+
+  // We assume the only valid responses are 'YES' and 'NO'
+  const assertResponse = response => {
+    assert(
+      response === 'NO' || response === 'YES',
+      details`the answer ${q(response)} was not 'YES' or 'NO'`,
+    );
+    // Throw an error if the response is not valid, but do not
+    // complete the offer. We should allow the voter to recast their vote.
+  };
+
+  const voterHook = voterOfferHandle => {
+    const voter = harden({
+      /**
+       * Vote on a particular issue
+       * @param {string} response - 'YES' || 'NO'
+       */
+      vote: response => {
+        // Throw if the offer is no longer active, i.e. the user has
+        // completed their offer and the assets are no longer escrowed.
+        assert(
+          zcf.isOfferActive(voterOfferHandle),
+          details`the escrowing offer is no longer active`,
+        );
+
+        assertResponse(response);
+
+        // Record the response
+        if (offerHandleToResponse.has(voterOfferHandle)) {
+          offerHandleToResponse.set(voterOfferHandle, response);
+        } else {
+          offerHandleToResponse.init(voterOfferHandle, response);
+        }
+        return `Successfully voted '${response}'`;
+      },
+    });
+    return voter;
+  };
+
+  const expectedVoterProposal = harden({
+    give: { Assets: null },
+  });
+
+  const makeVoterInvite = () =>
+    zcf.makeInvitation(checkHook(voterHook, expectedVoterProposal), 'voter');
+
+  const expectedSecretaryProposal = harden({});
+
+  const secretaryHook = secretaryOfferHandle => {
+    // TODO: what if the secretary offer is no longer active?
```

Comment:
Isn't the most likely case that it has already been closed? I think it shouldn't allow `close()` twice, though it would be fine to offer an intermediate count that didn't close voting.

We should probably require that it can't be closed early by an exit condition. Are there any other ways that the offer could be inactive?